[v1.16.1] Panic in controller manager

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

Recent installation of 1.16 kubernetes using kubeadm, controller manager panics frequently

```golang
E0919 03:22:37.000231       1 runtime.go:78] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x38d8420), concrete:(*runtime._type)(0x4171300), asserted:(*runtime._type)(0x371f240), missingMethod:""} (interface conversion: interface {} is *v1.Secret, not string)
goroutine 1485 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x3a23160, 0xc0019b5350)
	/workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x3a23160, 0xc0019b5350)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
k8s.io/kubernetes/pkg/controller/bootstrap.(*TokenCleaner).processNextWorkItem(0xc000a8bc20, 0xc0000c5100)
	/workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/bootstrap/tokencleaner.go:152 +0x2c9
k8s.io/kubernetes/pkg/controller/bootstrap.(*TokenCleaner).worker(0xc000a8bc20)
	/workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/bootstrap/tokencleaner.go:140 +0x2b
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00140c370)
	/workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x54
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00140c370, 0x2540be400, 0x0, 0x45d201, 0xc000282840)
	/workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc00140c370, 0x2540be400, 0xc000282840)
	/workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by k8s.io/kubernetes/pkg/controller/bootstrap.(*TokenCleaner).Run
	/workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/bootstrap/tokencleaner.go:124 +0x1d0
panic: interface conversion: interface {} is *v1.Secret, not string [recovered]
	panic: interface conversion: interface {} is *v1.Secret, not string
```

**What you expected to happen**:

Expect controller manager to run with no panics

**How to reproduce it (as minimally and precisely as possible)**:

install 1.16 cluster using kubeadm

**Anything else we need to know?**:

None

**Environment**:
- Kubernetes version (use `kubectl version`): 1.16
- Cloud provider or hardware configuration: azure
- OS (e.g: `cat /etc/os-release`): ubuntu 18
- Kernel (e.g. `uname -a`): 5.0.0-1018-azure
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

--- a/pkg/controller/bootstrap/tokencleaner.go
+++ b/pkg/controller/bootstrap/tokencleaner.go
@@ -202,6 +202,11 @@ func (tc *TokenCleaner) evalSecret(o interface{}) {
 			klog.V(3).Infof("Error deleting Secret: %v", err)
 		}
 	} else if ttl > 0 {
-		tc.queue.AddAfter(o, ttl)
+		key, err := controller.KeyFunc(o)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return
+		}
+		tc.queue.AddAfter(key, ttl)
 	}
 }

--- a/pkg/controller/bootstrap/tokencleaner_test.go
+++ b/pkg/controller/bootstrap/tokencleaner_test.go
@@ -110,10 +110,11 @@ func TestCleanerExpiredAt(t *testing.T) {
 
 	secret := newTokenSecret("tokenID", "tokenSecret")
 	addSecretExpiration(secret, timeString(2*time.Second))
+	secrets.Informer().GetIndexer().Add(secret)
+	cleaner.enqueueSecrets(secret)
 	expected := []core.Action{}
 	verifyFunc := func() {
-		secrets.Informer().GetIndexer().Add(secret)
-		cleaner.evalSecret(secret)
+		cleaner.processNextWorkItem()
 		verifyActions(t, expected, cl.Actions())
 	}
 	// token has not expired currently

